File: MultiLanguageSandbox/src/test/deps/c/12.c

#include <assert.h>
#include <math.h>
#include <stdbool.h>
#include <stdio.h>
/*
You are given three digits a, b, c. Two of them are equal, but the third one is different from the other two. Find the value that occurs exactly once.
    >>> extraNumber(0,0,1)
    1
    >>> extraNumber(4,3,4)
    3
*/
int extraNumber(int a, int b, int c) {
    if (a == b) {
        return c;
    } else if (a == c) {
        return b;
    } else {
        return a;
    }
}
int main()
{
    assert(extraNumber(2, 7, 2) == 7);
    assert(extraNumber(3, 2, 2) == 3);
    assert(extraNumber(5, 5, 1) == 1);
    assert(extraNumber(500000000, 3, 500000000) == 3);
    assert(extraNumber(500000000, 500000000, 3) == 3);
    return 0;
}

File: MultiLanguageSandbox/src/test/deps/pascal/12.pas
{$ASSERTIONS ON}

uses
  SysUtils, Math;
(*
Given three integers a, b, and c, with two of them being equal and the third one being distinct,
the task is to identify and return the distinct integer that occurs exactly once.
Examples:
>>> extraNumber(0,0,1) returns 1
>>> extraNumber(4,3,4) returns 3
*)

function extraNumber(a, b, c: integer): integer;
begin
    if a = b then
        extraNumber := c
    else if a = c then
        extraNumber := b
    else
        extraNumber := a;
end;
procedure testExtraNumber;
begin
    Assert(extraNumber(2, 7, 2) = 7);
    Assert(extraNumber(3, 2, 2) = 3);
    Assert(extraNumber(5, 5, 1) = 1);
    Assert(extraNumber(500000000, 3, 500000000) = 3);
    Assert(extraNumber(500000000, 500000000, 3) = 3);
end;

begin
    testExtraNumber;
    WriteLn('All test cases passed!');
end.